core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -790,25 +798,28 @@ public MergeWriterNode plan(Merge merge)
             }
 
             ImmutableList.Builder<Expression> rowBuilder = ImmutableList.builder();
+            Assignments.Builder checkConstraintsAssignments = Assignments.builder();
             List<ColumnHandle> mergeCaseSetColumns = mergeCaseColumnsHandles.get(caseNumber);
             for (ColumnHandle dataColumnHandle : mergeAnalysis.getDataColumnHandles()) {
                 int index = mergeCaseSetColumns.indexOf(dataColumnHandle);
+                int fieldNumber = mergeAnalysis.getColumnHandleFieldNumbers().get(dataColumnHandle);
+                Symbol field = targetTablePlan.getFieldMappings().get(fieldNumber);
```

Comment:
Is the usage of `targetTablePlan` instead of `planWithPresentColumn` intentional?